Sources/LanguageServerProtocol/CompletionItem.swift

```diff
@@ -13,70 +13,66 @@
+      public var detail: String?
 
-  /// The name to use for sorting the result. If `nil`, use `label.
-  public var sortText: String?
+      /// The name to use for sorting the result. If `nil`, use `label.
+      public var sortText: String?
 
-  /// The name to use for filtering the result. If `nil`, use `label.
-  public var filterText: String?
+      /// The name to use for filtering the result. If `nil`, use `label.
+      public var filterText: String?
 
-  /// The (primary) edit to apply to the document if this completion is accepted.
-  ///
-  /// This takes precedence over `insertText`.
-  ///
-  /// - Note: The range of the edit must contain the completion position.
-  public var textEdit: TextEdit?
+      /// The (primary) edit to apply to the document if this completion is accepted.
+      ///
+      /// This takes precedence over `insertText`.
+      ///
+      /// - Note: The range of the edit must contain the completion position.
+      public var textEdit: TextEdit?
 
-  /// **Deprecated**: use `textEdit`
-  ///
-  /// The string to insert into the document. If `nil`, use `label.
-  public var insertText: String?
+      /// **Deprecated**: use `textEdit`
+      ///
+      /// The string to insert into the document. If `nil`, use `label.
+      public var insertText: String?
 
-  /// The format of the `textEdit.nextText` or `insertText` value.
-  public var insertTextFormat: InsertTextFormat?
+      /// The format of the `textEdit.nextText` or `insertText` value.
+      public var insertTextFormat: InsertTextFormat?
 
-  /// The kind of completion item - e.g. method, property.
-  public var kind: CompletionItemKind
+      /// The kind of completion item - e.g. method, property.
+      public var kind: CompletionItemKind
 
-  /// Whether the completion is for a deprecated symbol.
-  public var deprecated: Bool?
+      /// Whether the completion is for a deprecated symbol.
+      public var deprecated: Bool?
 
-  // TODO: remaining members
+      // TODO: remaining members
 
-  public init(
-    label: String,
-    detail: String? = nil,
-    sortText: String? = nil,
-    filterText: String? = nil,
-    textEdit: TextEdit? = nil,
-    insertText: String? = nil,
-    insertTextFormat: InsertTextFormat? = nil,
-    kind: CompletionItemKind,
-    deprecated: Bool? = nil)
-  {
-    self.label = label
-    self.detail = detail
-    self.sortText = sortText
-    self.filterText = filterText
-    self.textEdit = textEdit
-    self.insertText = insertText
-    self.insertTextFormat = insertTextFormat
-    self.kind = kind
-    self.deprecated = deprecated
-  }
+      public init(
+            label: String, detail: String? = nil, sortText: String? = nil,
+            filterText: String? = nil, textEdit: TextEdit? = nil,
+            insertText: String? = nil,
+            insertTextFormat: InsertTextFormat? = nil, kind: CompletionItemKind,
+            deprecated: Bool? = nil
```

Comment:
What happened to always formatting parameters vertically XOR horizontally?  This change is a big regression IMO.  I can see why folks might want it in some specific cases, but it makes declarations harder to read.

Reply:
That rule was discarded after I encouraged it.  It is much too restrictive and can cause unreadably long columnar declarations where you see, e.g.

```swift
     ...parameters...
     red: Float,
     green: Float,
     blue: Float,
     alpha: Float,
     ...parameters...
```

which should really be grouped together on a single line.

```swift
     ...parameters...
     red: Float, green: Float, blue: Float, alpha: Float,
     ...parameters...
```

This goes not just for extremely long parameter lists.  It's very common to find declarations where all the parameters fit on two lines, and there's a natural grouping that makes the declaration read well.
 
Because the new rules allow discretionary line breaks, we can expect people to add breaks where it's best for readability (including making parameter columns when that's best), but the tools aren't going to unwrap my carefully-organized declarations and turn them into long columns.

The reason the declaration you see here looks so bad is that for the purposes of this test, discretionary line breaks are being ignored, but that's not the intended configuration.

Reply:
Right—we decided to completely ignore existing line breaks for this test so that we could get an intentionally more "opinionated" diff, to make sure that keeping those line breaks didn't somehow hide other issues.

The final version will preserve them (it's configurable right now, but that's more for development than intended use).